source/common/version/api_version.cc

```diff
@@ -0,0 +1,63 @@
+#include "common/version/api_version.h"
+
+#include <string>
+
+#include "common/common/fmt.h"
+
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_split.h"
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+
+std::string
+ApiVersionInfo::apiVersionToString(const envoy::config::core::v3::ApiVersionNumber& version) {
+  return fmt::format("{}.{}.{}", version.version().major_number(), version.version().minor_number(),
+                     version.version().patch());
+}
+
+const envoy::config::core::v3::ApiVersionNumber& ApiVersionInfo::apiVersion() {
+  static const auto* result =
+      new envoy::config::core::v3::ApiVersionNumber(makeApiVersion(API_VERSION_NUMBER));
+  return *result;
+}
+
+const envoy::config::core::v3::ApiVersionNumber& ApiVersionInfo::oldestApiVersion() {
+  static const auto* result =
+      new envoy::config::core::v3::ApiVersionNumber(computeOldestApiVersion(apiVersion()));
+  return *result;
+}
+
+envoy::config::core::v3::ApiVersionNumber ApiVersionInfo::makeApiVersion(const char* version) {
+  envoy::config::core::v3::ApiVersionNumber result;
+  // Split API_VERSION_NUMBER into version
+  const std::vector<std::string> ver_split = absl::StrSplit(version, '.');
+  if (ver_split.size() == 3) {
```

Comment:
Is there's a plan to validate the string in ` API_VERSION` file and maybe even fail the build if the format is wrong? IIUC this is important enough that a build without a valid api version shouldn't be possible?

Reply:
Yes, this validation will be part of the tooling around the API version management.